presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -63,6 +65,22 @@ public RuleAssert on(Function<PlanBuilder, PlanNode> planProvider)
         return this;
     }
 
+    public RuleAssert withStats(Map<PlanNodeId, PlanNodeStatsEstimate> stats)
+    {
+        setStats(plan, stats);
+        return this;
+    }
+
+    private void setStats(PlanNode node, Map<PlanNodeId, PlanNodeStatsEstimate> idStatsMap)
+    {
+        if (idStatsMap.containsKey(node.getId())) {
+            lookup.setStats(node, idStatsMap.get(node.getId()));
```

Comment:
it seems weird to set the stats in RuleTester for nodes that don't get created until RuleAssert.  what if I get rid of the Lookup parameter and build it in the constructor instead.  It doesn't get used in RuleTester anyway except  to pass it in to RuleAssert.